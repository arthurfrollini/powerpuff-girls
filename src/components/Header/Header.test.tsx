import { render, screen } from "@testing-library/react";
import { Header } from ".";

import renderer from "react-test-renderer";

const props = {
  isHome: false,
  imageSource:
    "https://upload.wikimedia.org/wikipedia/pt/0/08/The_Powerpuff_Girls_Logo.png",
  name: "Powerpuff Girls",
  summary:
    "As Meninas Superpoderosas é uma série de desenho animado criada e escrita por Craig McCracken. Sucesso em todo o mundo!",
};

describe("Header component", () => {
  it("renders correctly", () => {
    const { getByText, debug } = render(<Header {...props} />);

    const header = screen.getByTestId("header");

    expect(header).toBeTruthy();
    expect(getByText("Powerpuff Girls")).toBeInTheDocument();

    debug();
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Header {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
