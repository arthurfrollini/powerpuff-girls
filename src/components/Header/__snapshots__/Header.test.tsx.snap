// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div
  className="sc-bdvvtL gzBWRV"
  data-testid="header"
>
  <h1
    className="sc-dkPtRN dIVUCs"
  >
    Powerpuff Girls
  </h1>
  <div
    className="sc-eCImPb iKOuKf"
  >
    <img
      alt="PowerPuff Girls Image"
      className="sc-gKclnd klwHyn"
      height="200px"
      src="https://upload.wikimedia.org/wikipedia/pt/0/08/The_Powerpuff_Girls_Logo.png"
      style={
        Object {
          "marginBottom": 30,
        }
      }
    />
    <p
      className="sc-furwcr ehkaCT"
    >
      As Meninas Superpoderosas é uma série de desenho animado criada e escrita por Craig McCracken. Sucesso em todo o mundo!
    </p>
  </div>
</div>
`;
